src/extensions/lens-renderer-extension.ts

```diff
@@ -53,31 +60,35 @@ export class LensRendererExtension extends LensExtension {
    *
    * The default implementation is to return `true`
    */
-  async isEnabledForCluster(cluster: KubernetesCluster): Promise<Boolean> {
-    return (void cluster) || true;
+  isEnabledForCluster(cluster: KubernetesCluster): Promise<Boolean> {
+    return Promise.resolve((void cluster) || true);
   }
 
   /**
+   * @deprecated Just push to `.catalogFilters` instead
    * Add a filtering function for the catalog entities. This will be removed if the extension is disabled.
    * @param fn The function which should return a truthy value for those entities which should be kept.
    * @returns A function to clean up the filter
    */
   addCatalogFilter(fn: EntityFilter): Disposer {
-    const dispose = catalogEntityRegistry.addCatalogFilter(fn);
+    const dispose = once(() => this.catalogFilters.remove(fn));
 
+    this.catalogFilters.push(fn);
     this[Disposers].push(dispose);
 
     return dispose;
   }
 
   /**
+   * @deprecated Just push to `.catalogCategoryFilters` instead
    * Add a filtering function for the catalog categories. This will be removed if the extension is disabled.
    * @param fn The function which should return a truthy value for those categories which should be kept.
    * @returns A function to clean up the filter
    */
   addCatalogCategoryFilter(fn: CategoryFilter): Disposer {
-    const dispose = catalogCategoryRegistry.addCatalogCategoryFilter(fn);
+    const dispose = once(() => this.catalogCategoryFilters.remove(fn));
```

Comment:
Redundant `once`. Superficial.

Reply:
Not really, I think it helps prevent bugs from the extensions side.